rpcserver_test.go

```diff
@@ -0,0 +1,177 @@
+		accMgr *account.MockManager, marshalerMock *MockMarshaler,
+	) {
+		// Renew account params
+		bestHeight := uint32(100)
+		feeRate := chainfee.SatPerKWeight(req.FeeRateSatPerKw)
+		expiryHeight := req.GetAbsoluteExpiry()
+		if expiryHeight == 0 {
+			expiryHeight = 100 + req.GetRelativeExpiry()
+		}
+		// RenewAccount returns
+		acc := &account.Account{}
+		tx := &wire.MsgTx{}
+		accMgr.EXPECT().
+			RenewAccount(
+				gomock.Any(), getAccountKey(req.AccountKey),
+				expiryHeight, feeRate, bestHeight,
+			).
+			Return(acc, tx, nil)
+
+		rpcAccount := poolrpc.Account{}
+		marshalerMock.EXPECT().
+			MarshallAccountsWithAvailableBalance(
+				gomock.Any(), gomock.Eq([]*account.Account{acc}),
+			).
+			Return([]*poolrpc.Account{&rpcAccount}, nil)
+	},
+	checkResponse: func(*poolrpc.RenewAccountResponse) error {
+		return nil
+	},
+}, {
+	name: "account key must be valid",
+	getReq: func() *poolrpc.RenewAccountRequest {
+		return &poolrpc.RenewAccountRequest{
+			AccountKey: []byte{3, 5, 8},
+		}
+	},
+	expectedError: "invalid pub key length 3",
+	mockSetter: func(req *poolrpc.RenewAccountRequest,
+		accMgr *account.MockManager, marshalerMock *MockMarshaler,
+	) {
+	},
+	checkResponse: func(*poolrpc.RenewAccountResponse) error {
+		return nil
+	},
+}, {
+	name: "req should specify absolute/relative expiry",
+	getReq: func() *poolrpc.RenewAccountRequest {
+		return genRenewAccountReq(traderKeyRaw, 0, 0, 1000)
+	},
+	expectedError: "either relative or absolute height must be specified",
+	mockSetter: func(req *poolrpc.RenewAccountRequest,
+		accMgr *account.MockManager, marshalerMock *MockMarshaler,
+	) {
+	},
+	checkResponse: func(*poolrpc.RenewAccountResponse) error {
+		return nil
+	},
+}, {
+	name: "req should specify a valid fee rate",
+	getReq: func() *poolrpc.RenewAccountRequest {
+		return genRenewAccountReq(traderKeyRaw, 0, 100, 0)
+	},
+	expectedError: "fee rate of 0 sat/kw is too low, minimum is 253 sat/kw",
+	mockSetter: func(req *poolrpc.RenewAccountRequest,
+		accMgr *account.MockManager, marshalerMock *MockMarshaler,
+	) {
+	},
+	checkResponse: func(*poolrpc.RenewAccountResponse) error {
+		return nil
+	},
+}}
+
+func TestRenewAccount(t *testing.T) {
+	for _, tc := range renewAccountTestCases {
+		tc := tc
+
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			mockCtrl := gomock.NewController(t)
+			defer mockCtrl.Finish()
+
+			req := tc.getReq()
+
+			accountMgr := account.NewMockManager(mockCtrl)
+			orderMgr := order.NewMockManager(mockCtrl)
+			marshaler := NewMockMarshaler(mockCtrl)
+			tc.mockSetter(req, accountMgr, marshaler)
+
+			srv := rpcServer{
+				accountManager: accountMgr,
+				orderManager:   orderMgr,
+				marshaler:      marshaler,
+			}
+			srv.bestHeight = 100
+
+			ctx, cancel := context.WithTimeout(
+				context.Background(), ctxTimeout,
+			)
+			defer cancel()
+
+			resp, err := srv.RenewAccount(ctx, req)
+			if tc.expectedError != "" {
+				assert.EqualError(t, err, tc.expectedError)
```

Comment:
nit: `require` and the expectation comes first iirc

Reply:
It looks like for `EqualError` this is the right order? 

https://github.com/stretchr/testify/blob/master/assert/assertions.go#L1367